src/module.c

```diff
@@ -826,23 +828,155 @@ int TSDB_incrby(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
     return rv;
 }
 
+int _tsdb_get_block_callback(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+    REDISMODULE_NOT_USED(argv);
+    REDISMODULE_NOT_USED(argc);
+
+    RedisModuleString *keyname = RedisModule_GetBlockedClientReadyKey(ctx);
+
+    Series *series;
+    RedisModuleKey *key;
+    const int status = GetSeries(ctx, keyname, &key, &series, REDISMODULE_READ);
+    if (!status) {
+        return REDISMODULE_ERR;
+    }
+    api_timestamp_t *timestamp = RedisModule_GetBlockedClientPrivateData(ctx);
+
+    if (*timestamp < series->lastTimestamp) {
```

Comment:
@gkorland maybe there's more than 1 sample? we want to return the first one after the timestamp?

Reply:
`TS.GET` always returns one event